tensorflow/core/kernels/depthwise_conv_op_gpu.cu.cc

```diff
@@ -1153,8 +1154,13 @@ __launch_bounds__(1024, 2) void DepthwiseConv2dBackpropFilterGPUKernelNHWCSmall(
       if (filter_depth < in_depth) {
         T val = accum_data[i];
         // Warp-accumulate the pixels of the same depth from the accumulator.
+        int lane_id;
+        asm volatile ("mov.u32 %0, %laneid;" : "=r"(lane_id));
+        int sub_warp = lane_id / kAccumPixels;
+        int zeros = sub_warp * kAccumPixels;
+        unsigned mask = ((1U << kAccumPixels) - 1) << zeros;
         for (int delta = kAccumPixels / 2; delta > 0; delta /= 2) {
-          val += CudaShuffleDown(val, delta);
+          val += CudaShuffleXor(mask, val, delta);
         }
```

Comment:
This whole chunk of code is now a lot more complicated than it was, and is repeated elsewhere.  Can we abstract it into a function?

Reply:
Available in cub I believe

Reply:
Done.